### ISSUE TITLE
async_server_base::response typedef no longer available?

### ISSUE THREAD

**Issue by igorpeshansky** [cpp-netlib/cpp-netlib#829]
The `async_server_base::response` typedef seems to have been [removed](https://github.com/cpp-netlib/cpp-netlib/commit/65b155115e442a7ae28636399e708ddf91833084#diff-83a8552bfbb515eba3d98a01a3fb6c6eL25) in 65b155115e442a7ae28636399e708ddf91833084 as part of the stale documentation cleanup. Was this an accident, or was the removal intentional? If the latter, what's the recommended way to access the server's response type?
@deanberris FYI.

**Comment by igorpeshansky** [comment]
Never mind, looks like the response is not exposed in the new API. The handler callback has access to the connection object, and the API is now:
* 400 response:
```cpp
using http_server = http::server<handler_type>;
void Handler(const http_server::request& request, std::shared_ptr<http_server::connection> conn) {
  conn->set_status(http_server::connection::bad_request);
}
```
* 200 response with data:
```cpp
using http_server = http::server<handler_type>;
void Handler(const http_server::request& request, std::shared_ptr<http_server::connection> conn) {
  conn->set_status(http_server::connection::ok);
  conn->write("Some data");
}
```
Posting here because I couldn't find this in any of the [latest docs](http://cpp-netlib.org/0.13.0/), which still refer to the old API.

**Comment by deanberris** [comment]
Right, thanks @igorpeshansky -- we should update the documentation, definitely.

**Comment by igorpeshansky** [comment]
Note that we found that this new API makes such handlers really hard to test — where previously one could examine the response object, there are no accessors into the connection state.

**Comment by deanberris** [comment]
I suspect we can improve the connection type, to expose some useful state information about the response. The problem with the previous API was that it required the data to be complete before the handler was invoked. This API allows the handler to handle the data incrementally. This tradeoff may have gone too far one direction, and we could really improve this a lot more by allowing handlers to choose which way to go.

Do you have thoughts on how we might be able to make this better, in terms of testability?
